cognitedata/reveal
# fix: primitive buffer parsing with multiple types in same buffer

#### Type of change

![Bug](https://img.shields.io/badge/Type-Bug-red) 

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/

## Description :pencil:


The current version of the primitive parsing utility expects only one primitive type to be present in the input buffer. This fixes it by making sure we start reading primitives at the correct offset.

In addition, it imports `RevealGeometryCollectionType` directly from its source file instead of from the sector parser index file, in order to prevent a circular dependency (or some other kind of error, not sure).

Suggestion - perhaps the primitive parsing utility should be in the `sector-parser` package instead? In that case, it should probably sit in a `/test` folder, since it's not used in the production code.

Primitive parsing is only used for tests and debugging, so not really a user facing bug fix, but oh well.

## How has this been tested? :mag:




- Apply below diff
- Start examples
- When `primitives` load, observe that all general cylinders in the scene are printed with reasonable parameters and tree indices in the console. They were not before.

[parserDiff.txt](https://github.com/cognitedata/reveal/files/10906478/parserDiff.txt)

## Test instructions :information_source:




## Checklist :ballot_box_with_check:
